src/xpu/ocl/utils.hpp

```diff
@@ -74,6 +74,84 @@ const char *convert_cl_int_to_str(cl_int cl_status);
         MAYBE_UNUSED(s); \
     } while (false)
 
+#if defined(_WIN32)
+#define OCL_LIB_NAME "OpenCL.dll"
+#elif defined(__linux__)
+#define OCL_LIB_NAME "libOpenCL.so.1"
+#endif
+
+template <typename F>
+F find_ocl_symbol(const char *symbol) {
+    auto f = (F)xpu::find_symbol(OCL_LIB_NAME, symbol);
+    if (!f) VERROR(common, ocl, "cannot find symbol: %s", symbol);
+    return f;
+}
+#undef OCL_LIB_NAME
+
+#define INDIRECT_OCL_CALL(result_type, f) \
+    template <typename... Args> \
+    result_type f(Args &&...args) { \
+        static auto f_ = find_ocl_symbol<decltype(&::f)>(#f); \
+        return f_(std::forward<Args>(args)...); \
+    }
+INDIRECT_OCL_CALL(cl_int, clBuildProgram)
```

Comment:
Since OpenCL linking is now indirect, would it make sense to enable the OCL GPU runtime by default?